From 3e4255ae15ac4b4f87b7fc94f4e051479bc1a05b Mon Sep 17 00:00:00 2001
From: Takumi Ando <takumi@spacecubics.com>
Date: Thu, 2 Jul 2026 19:03:21 +0900
Subject: [PATCH 1/2] mtd: spi-nor: Add Avalanche Dual Quad MRAM support

Add Avalanche Dual Quad MRAM manufacturer ID detection and configuration.

Signed-off-by: Takumi Ando <takumi@spacecubics.com>
---
 drivers/mtd/spi/Kconfig       |  5 +++++
 drivers/mtd/spi/spi-nor-ids.c | 10 ++++++++++
 2 files changed, 15 insertions(+)

diff --git a/drivers/mtd/spi/Kconfig b/drivers/mtd/spi/Kconfig
index 48fc09d2b33..e0e464daf53 100644
--- a/drivers/mtd/spi/Kconfig
+++ b/drivers/mtd/spi/Kconfig
@@ -250,6 +250,11 @@ config SPI_FLASH_ZBIT
 	help
 	  Add support for Zbit Semiconductor Inc. SPI flash chips (ZB25xxx).
 
+config SPI_FLASH_AVALANCHE
+	bool "Avalanche Dual QSPI MRAM support"
+	help
+	  Add support for Avalanche Dual QSPI MRAM chips (AS30xG208).
+
 endif
 
 config SPI_FLASH_USE_4K_SECTORS
diff --git a/drivers/mtd/spi/spi-nor-ids.c b/drivers/mtd/spi/spi-nor-ids.c
index d6d27bc3857..21210b28b2f 100644
--- a/drivers/mtd/spi/spi-nor-ids.c
+++ b/drivers/mtd/spi/spi-nor-ids.c
@@ -77,6 +77,16 @@ const struct flash_info spi_nor_ids[] = {
 	{ INFO("at25sl321",	0x1f4216, 0, 64 * 1024,  64, SECT_4K) },
 	{ INFO("at26df081a",	0x1f4501, 0, 64 * 1024,  16, SECT_4K) },
 #endif
+#ifdef CONFIG_SPI_FLASH_AVALANCHE	/* AVALANCHE */
+	{ INFO("as301g208", 0xe62128, 0, 64 * 1024, 1024,
+		SECT_4K | USE_FSR | SPI_NOR_QUAD_READ | NO_CHIP_ERASE) },
+	{ INFO("as302g208", 0xe62129, 0, 64 * 1024, 2048,
+		SECT_4K | USE_FSR | SPI_NOR_QUAD_READ | NO_CHIP_ERASE) },
+	{ INFO("as304g208", 0xe6212a, 0, 64 * 1024, 4096,
+		SECT_4K | USE_FSR | SPI_NOR_QUAD_READ | NO_CHIP_ERASE) },
+	{ INFO("as308g208", 0xe6212c, 0, 64 * 1024, 8192,
+		SECT_4K | USE_FSR | SPI_NOR_QUAD_READ | NO_CHIP_ERASE) },
+#endif
 #ifdef CONFIG_SPI_FLASH_EON		/* EON */
 	/* EON -- en25xxx */
 	{ INFO("en25q80b",   0x1c3014, 0, 64 * 1024,   16, SECT_4K) },
-- 
2.54.0

